mono/metadata/w32file.c

```diff
@@ -446,7 +446,7 @@ incremental_find_check_match (IncrementalFind *handle, WIN32_FIND_DATA *data, Mo
 	
 	full_name = g_build_filename (handle->utf8_path, utf8_result, NULL);
 	g_free (utf8_result);
-	*result = mono_string_new_checked (mono_domain_get (), full_name, error);
+	*result = mono_string_new_handle (mono_domain_get (), full_name, error);
```

Comment:
This allocates a new handle every time `incremental_find_check_match` is called which will accumulate until it's caller pops a handle stack frame.   Better to wrap `incremental_find_check_match` in `HANDLE_FUNCTION_{ENTER,RETURN_VAL}` (and make it single-return).